apis/v1beta1/vspherevm_webhook.go

```diff
@@ -107,6 +107,10 @@ func (r *VSphereVM) ValidateUpdate(old runtime.Object) error {
 		delete(newVSphereVMSpec, "os")
 	}
 
+	// allow changes to thumbprint
+	delete(oldVSphereVMSpec, "thumbprint")
+	delete(newVSphereVMSpec, "thumbprint")
```

Comment:
Since we have the same `delete` call being repeated over and over, consider creating a separate method which takes in a spec and a key and removes it from the spec map. List the keys which are allowed updates in an array and for each one pass it to this new function.

Reply:
Sure, I can refine them in next patch.